### PR TITLE
Small bugfixes and OLM changes to be able to publish it in a registry

### PR DESCRIPTION
**Pull request by vboulineau**
### What does this PR do?

I did not document the OLM process but we should, though it will change with operator-sdk 0.16 (that we'll to migrate to have the zap option).

I've tried the OLM workflow e2e, pushing to Quay.io and adding it as a custom source in OpenShift (https://quay.io/application/vboulineau/datadog-operator)

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: deploy/olm-catalog/datadog-operator/0.1.1/datadog-operator.v0.1.1.clusterserviceversion.yaml [hunk 1]
```diff
@@ -53,7 +77,7 @@ spec:
                       fieldPath: metadata.name
                 - name: OPERATOR_NAME
                   value: datadog-operator
-                image: datadog/operator:latest
+                image: datadog/operator:0.1.1
```
Comment (clamoriniere) [inline deploy/olm-catalog/datadog-operator/0.1.1/datadog-operator.v0.1.1.clusterserviceversion.yaml]: can we still use "latest" here?

Comment (vboulineau) [reply deploy/olm-catalog/datadog-operator/0.1.1/datadog-operator.v0.1.1.clusterserviceversion.yaml]: IMHO, I believe we should correlate CSV version and image version, we might be do code changes that require changes in the CSV. As OLM is meant to have auto-upgrade, I believe we should always bundle the release.

Comment (clamoriniere) [reply deploy/olm-catalog/datadog-operator/0.1.1/datadog-operator.v0.1.1.clusterserviceversion.yaml]: agree, but for now we don't have this image released

Comment (clamoriniere) [reply deploy/olm-catalog/datadog-operator/0.1.1/datadog-operator.v0.1.1.clusterserviceversion.yaml]: we should generate the OLM when we release the a new operator version

Comment (clamoriniere) [reply deploy/olm-catalog/datadog-operator/0.1.1/datadog-operator.v0.1.1.clusterserviceversion.yaml]: We can stay with 0.1.1 and don't forget to deliver the next version with this value

Comment (vboulineau) [reply deploy/olm-catalog/datadog-operator/0.1.1/datadog-operator.v0.1.1.clusterserviceversion.yaml]: I do have pushed a 0.1.1 for my tests

**Comment by vboulineau** [comment]
Added fix for log Volumes

**Comment by clamoriniere** [comment]
could you rebase your PR?

File: deploy/clusterrole.yaml [hunk 2]
```diff
@@ -12,3 +12,9 @@ rules:
   - 'clusterrolebindings'
   verbs:
   - '*'
+- apiGroups:
+  - datadoghq.com
+  resources:
+  - datadogagents/finalizers
```
Comment (clamoriniere) [inline deploy/clusterrole.yaml]: I show it also in the "Role": https://github.com/DataDog/datadog-operator/pull/35/files#diff-4374306bd7fd15cc09428b7ccf9bbec8R53

Comment (vboulineau) [reply deploy/clusterrole.yaml]: Yep we should align there

**Pull request merged**